`azurerm_kubernetes_cluster` - support for `upgrade_override_setting` property (#27962)
hashicorp/terraform-provider-azurerm: website/docs/r/kubernetes_cluster.html.markdown

---

The `upgrade_override` block supports the following:

-> **Note:** Once set, the `upgrade_override` block cannot be removed from the configuration.

* `force_upgrade_enabled` - (Required) Whether to force upgrade the cluster. Possible values are `true` or `false`.

> !> **Important: The `force_upgrade_enabled` field instructs the upgrade operation to bypass upgrade protections (e.g. checking for deprecated API usage) which may render the cluster inoperative after the upgrade process has completed. Use the `force_upgrade_enabled` option with extreme caution only.**

Reviewer: Per the new [contributor docs guidelines](https://github.com/hashicorp/terraform-provider-azurerm/pull/28611/files) around note annotation we switch this up to the below:

```suggestion
!> **Note:** The `force_upgrade_enabled` field instructs the upgrade operation to bypass upgrade protections (e.g. checking for deprecated API usage) which may render the cluster inoperative after the upgrade process has completed. Use the `force_upgrade_enabled` option with extreme caution only.
```

The `upgrade_override` block supports the following:

-> **Note:** Once set, the `upgrade_override` block cannot be removed from the configuration.

* `force_upgrade_enabled` - (Required) Whether to force upgrade the cluster. Note that this option instructs the upgrade operation to bypass upgrade protections such as checking for deprecated API usage.

> -> **Note: Enable this option only with caution as it may render the cluster inoperative.**

Reviewer: Maybe update the text to be something like:
```suggestion
!> **Important: The `force_upgrade_enabled` field instructs the upgrade operation to bypass upgrade protections (e.g. checking for deprecated API usage) which may render the cluster inoperative after the upgrade process has completed. Use the `force_upgrade_enabled` option with extreme caution only.**
```

> * `force_upgrade_enabled` - (Required) Whether to force upgrade the cluster. Note that this option instructs the upgrade operation to bypass upgrade protections such as checking for deprecated API usage.

Reviewer: So sorry I missed this bit in the last review... DOH! 
```suggestion
* `force_upgrade_enabled` - (Required) Whether to force upgrade the cluster. Possible values are `true` or `false`.
```
